File: apple/Sources/ReactNativeWatchOS/Modifiers/TextInputAutocapitalization.swift
import SwiftUI

/// SwiftUI `.textInputAutocapitalization(_:)`. Controls how text entry
/// auto-capitalizes. `TextInputAutocapitalization` and this modifier are
/// watchOS 9+, which equals the deployment target, so no availability gate is
/// required. An unrecognized `value` leaves the view unchanged.
enum RNWTextInputAutocapitalizationModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("textInputAutocapitalization") { view, params, _ in
            guard let value = parse(params.string("value")) else { return view }
            return AnyView(view.textInputAutocapitalization(value))
        }
    }

    private static func parse(_ s: String?) -> TextInputAutocapitalization? {
        switch s {
        case "never":      return .never
        case "words":      return .words
        case "sentences":  return .sentences
        case "characters": return .characters
        default:           return nil
        }
    }
}

File: src/modifiers/textInputAutocapitalization.ts
import { createModifier } from './createModifier';

/// SwiftUI `TextInputAutocapitalization` values.
export type TextInputAutocapitalizationKind =
  | 'never'
  | 'words'
  | 'sentences'
  | 'characters';

export interface TextInputAutocapitalizationParams {
  value: TextInputAutocapitalizationKind;
}

/// SwiftUI `.textInputAutocapitalization(_:)` (watchOS 9+). Controls how text
/// entry auto-capitalizes. Pass a value string directly or an object. No-ops
/// on watchOS versions earlier than 9.
export function textInputAutocapitalization(
  value: TextInputAutocapitalizationKind
): ReturnType<typeof createModifier>;
export function textInputAutocapitalization(
  params: TextInputAutocapitalizationParams
): ReturnType<typeof createModifier>;
export function textInputAutocapitalization(
  arg: TextInputAutocapitalizationKind | TextInputAutocapitalizationParams
) {
  if (typeof arg === 'string') {
    return createModifier('textInputAutocapitalization', { value: arg });
  }
  return createModifier('textInputAutocapitalization', arg);
}
